pyproject.toml

```diff
@@ -0,0 +1,97 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "avocado-framework-plugin-vt"
+dynamic = ["version"]
+description = "Avocado Plugin for Virtualization Testing"
+readme = {file = "README.rst", content-type = "text/x-rst"}
+license = {file = "LICENSE"}
+requires-python = ">=3.9"
+authors = [
+    { name = "Avocado Developers", email = "avocado-devel@redhat.com" }
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)",
+    "Natural Language :: English",
+    "Operating System :: POSIX",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Software Development :: Quality Assurance",
+    "Topic :: Software Development :: Testing",
+]
+dependencies = [
+    "netifaces",
+    "packaging",
+    "six",
+    "aexpect",
+    "avocado-framework>=82.1",
```

Comment:
Let's check would require a higher version of Avocado due to the python compatibility.

Reply:
@richtja What is the situation with a formal pyproject.toml support on the Avocado side? @PaulYuuu I already have one avocado plugin that has been fully migrated and had no problem installing it and using it with avocado on an as-is basis. So I suspect Avocado is not a blocker for this migration in any way.

Reply:
@pevogam Yes, that should be right. but considering this is a big milestone, so let's consider it a little more, we should also clarify which version we want to support after switching to pyproject.

About avocado, it's not a easy task. the main setup.py contains many logic to support optional plugins, we need to decouple them. I've thought optional-dependencies probably a right implementation and I have asked deepwiki to check if this is possible. You can have a look of this: https://deepwiki.com/search/avocado-is-using-setuppy-i-hop_4fd8c6a1-b483-462b-ac49-e3e25367ef40